Quaternion/Quaternion.py

```diff
@@ -156,21 +158,21 @@ def __init__(self, attitude=None, transform=None, q=None, equatorial=None):
         # checking correct shapes
         if q is not None:
             q = np.atleast_1d(q)
-            self._shape = q.shape[:-1]
+            self.shape = q.shape[:-1]
```

Comment:
Making this public is a great idea!  Maybe it should be a read-only property in case someone thinks they can reshape by setting `shape` the way you can in numpy.

Reply:
right, it should be read-only.

After I started this, I began wondering how much "array" interface this class should have, although what we have now is the minimal approach and should limit the scope of this merge.

Examples of array interface can include:
- the ability to multiply/divide/dx scalar quaternions by multi-quaternions (not allowed right now)
- the ability to iterate over quaternions in a multi-quaternion.
- broadcasting to multiply/divide/dx arbitrarily-sized multi-quaternions.

This will not be done for this merge, but you can think about it.